src/notation/internal/notationinteraction.cpp

```diff
@@ -1389,7 +1394,7 @@ void NotationInteraction::applyDropPaletteElement(Ms::Score* score, Ms::Element*
         }
         dropData.dropElement = 0;
 
-        notifyAboutDropChanged();
```

Comment:
This was the main culprit - calling this before apply() means the cmdState gets reset and hence endCmd() doesn't update the score as required.
I'm still not convinced this is quite right though as now there are two serialized calls to startCmd()/endCmd() for each add-from-palette operation (previously they were nested), which might cause an issue with undo - but the undo/redo stack is already unstable in MU4 so hard to know if it's caused an issue.